fern/01-guide/08-frameworks/01-react-nextjs/01-app-router.mdx

```diff
@@ -0,0 +1,452 @@
+---
+title: Next.js App Router Setup
+---
+
+This guide walks you through setting up BAML with Next.js App Router, leveraging Server Actions and React Server Components for optimal performance.
+
+## Quick Start
+
+Follow the step-by-step instructions below to set up BAML in a new or existing Next.js project.
+
+<Steps>
+### Create a New Next.js Project
+
+First, create a new Next.js project with the App Router:
+
+<CodeBlocks>
+```bash npm
+npx create-next-app@latest my-baml-app
+```
+
+```bash pnpm
+pnpm create next-app my-baml-app
+```
+
+```bash yarn
+yarn create next-app my-baml-app
+```
+</CodeBlocks>
+
+When prompted, make sure to:
+- Select **Yes** for "Would you like to use TypeScript?"
+- Select **Yes** for "Would you like to use the App Router? (recommended)"
+- Configure other options as needed for your project
+
+### Install Dependencies
+
+Next, install BAML and its dependencies:
+
+<CodeBlocks>
+```bash npm
+npm install @boundaryml/baml
+npm install @boundaryml/baml-nextjs-plugin
```

Comment:
```suggestion
npm install @boundaryml/baml @boundaryml/baml-nextjs-plugin
```

and I think we can do the same for the others too?

Reply:
Done